### PR TITLE
Migrate to pyproject.toml

### PR DESCRIPTION
**Pull request by hallvictoria**
- removed setup.py and added pyproject.toml for base and blob extensions
- added pre-commit hooks with black and isort similarly to worker

### CODE REVIEW

**Review by YunchuWang — APPROVED** [review]

File: azure-functions-extension-base/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        'Environment :: Web Environment',
+        'Development Status :: 5 - Production/Stable',
+    ]
+
+[project.optional-dependencies]
+dev = [
+        'mypy',
+        'pytest',
+        'pytest-cov',
+        'requests==2.*',
```
Comment (gavin-aguiar) [inline azure-functions-extension-base/pyproject.toml]: Do we need requests?

File: azure-functions-extension-blob/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        'Development Status :: 5 - Production/Stable',
+    ]
+dependencies = [
+    'azure-functions-extension-base',
+    'azure-storage-blob==12.19.0'
+    ]
+
+[project.optional-dependencies]
+dev = [
+    'mypy',
```
Comment (gavin-aguiar) [inline azure-functions-extension-blob/pyproject.toml]: Do we have a mypy tests? If not remove this,

Comment (gavin-aguiar) [reply azure-functions-extension-blob/pyproject.toml]: same for all pipelines

**Review by gavin-aguiar — APPROVED** [review]

File: azure-functions-extension-base/pyproject.toml (hunk 1 above)
Comment (hallvictoria) [reply azure-functions-extension-base/pyproject.toml]: Nope, removed

File: azure-functions-extension-blob/pyproject.toml (hunk 2 above)
Comment (hallvictoria) [reply azure-functions-extension-blob/pyproject.toml]: Removed

**Pull request merged**